lib/liquid/block.ex

```diff
@@ -12,6 +12,18 @@ defmodule Liquid.Block do
   alias Liquid.Tag, as: Tag
   alias Liquid.Block, as: Block
 
+  @type t :: %Liquid.Block{
+          name: String.t() | nil,
+          markup: String.t() | nil,
+          condition: String.t() | nil,
+          parts: [],
```

Comment:
Is parts etc always an empty list, or a list?

Reply:
it is a list of parts

Reply:
Then shouldn't is at least be `parts: [...] | []` or `parts: list()`, and the same for the other properties. The type now specifies that those are always empty lists